Scripts/Scarlet Enclave: Death Comes From On High

DanVS opened the pull request:
**Changes proposed:**
Possible fix for quest [Death Comes From On High (12641)](https://woehead.way-of-elendil.fr/?quest=12641). Schedule launch event on initialization stage.

**Target branch(es):** 3.3.5

**Issues addressed:** Closes #20843

**Tests performed:** Does build and works in-game.

DanVS committed:
schedule event on init

DanVS committed:
schedule event on init

Aokromes commented on src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:
```diff
@@ -455,6 +448,7 @@ class npc_eye_of_acherus : public CreatureScript
         }
 };
 
+
```
remove :)

DanVS replied:
Fixed in less then 1 minute.

DanVS committed:
whitespaces

DanVS committed:
whitespaces

DanVS committed:
fix model

DanVS committed:
fix model

Aokromes commented on sql/updates/world/3.3.5/9999_99_99_99_world.sql:
```diff
@@ -0,0 +1,2 @@
+-- Eye of Acherus (28511)
+UPDATE `creature_template` SET `modelid1`=26320, `modelid2`=0 WHERE `entry`=28511;
```
this is hack without sniffs.

DanVS replied:
Restored previous hack.

DanVS committed:
rolling back

DanVS committed:
rolling back

Aokromes commented on src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:
```diff
@@ -376,16 +376,18 @@ class npc_eye_of_acherus : public CreatureScript
-                if (Player* owner = me->GetCharmerOrOwner()->ToPlayer())
-                    owner->SendAutoRepeatCancel(me);
-
+            	me->SetDisplayId(me->GetCreatureTemplate()->Modelid1);
```
1 space more :)

DanVS replied:
Fixed :+1:

DanVS committed:
spaces

DanVS committed:
spaces

illfated commented:
@DanVS : title suggestion: **Scripts/Scarlet Enclave: Death Comes From On High**
"Core" and "Scripts" are generally 2 separate main categories, like Core/Spells vs Scripts/Spells.
When it comes to C++ scripts, it is usually **Scripts/** + area name or dungeon name.

ZenoX92 committed:
DB/Quest: Quest Emotes (Redridge Mountains)

Closes #22056

DanVS committed:
DB/Creature: Scarlet Crusader

Closes #22055

ZenoX92 committed:
DB/Quest: Investigate the Alchemist Shop / Blue Recluse

Closes #18997

Jildor committed:
TheSlavePens/Ahune: Fix crash (#22070)

Closes: #22069

sirikfoll committed:
Scripts/TheSlavePens: Boss Ahune fixes and updates

Fixes some visual problems
Fixes encounter start
Updated codestyle

Closes #22071

adeutscher committed:
Flag Explosive Shot as a negative spell effect. Closes #21922. (#22045)

Ovahlord committed:
Core/Player: fix periodic eating and drinking emotes (#21280)

Closes: #12567

Gooyeth committed:
Common: Replace ip2nation by ip2location. (#21957)

Replace ip2nation by ip2location.

Download: https://lite.ip2location.com/database/ip-country

Gooyeth committed:
Typo fix contry -> country (#22075)

Jildor committed:
Nopch fix (#22077)

ZenoX92 committed:
DB /Quest: Jungle Secrets

Closes #17286

ZenoX92 committed:
DB/Creature: Forlorn Spirit

Closes #18202

Jonne733 committed:
DB/Quest: More Terokkar Forest quest emotes

Closes #22076

ZenoX92 committed:
DB/Creature: Bookie Herod

Closes #22072

ZenoX92 committed:
DB/SAI: Duskwood

Closes #22068

ZenoX92 committed:
DB/Quest: Quest Emotes (Duskwood)

Closes #22065

Jonne733 committed:
DB/Creature: Lieutenant Gravelhammer and Captain Auric Sunchaser event

Closes #22044

Alpharius committed:
DB/Quests: DE locales for westfall, stormwind, redridge

Closes #22079

Killyana committed:
DB: Typo

Shauren committed:
Core/Misc: Improved ip2location code and set it to disabled by default

Killyana committed:
DB/Quest: Move Ulag the Cleaver to SAI

Keader commented on src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:
```diff
@@ -434,14 +435,7 @@ class npc_eye_of_acherus : public CreatureScript
-                    me->SetDisableGravity(true);
-                    me->SetControlled(true, UNIT_STATE_ROOT);
-                    _events.ScheduleEvent(EVENT_MOVE_START, 5000);
+                    me->CastSpell(me, SPELL_EYE_FLIGHT);
```
DoCastSelf

illfated replied:
... using the format
```cpp
DoCastSelf(SPELL_EYE_FLIGHT);
```
(if I recall correctly)

Keader commented on src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:
```diff
@@ -377,15 +377,17 @@ class npc_eye_of_acherus : public CreatureScript
-                if (Player* owner = me->GetCharmerOrOwner()->ToPlayer())
-                    owner->SendAutoRepeatCancel(me);
-
+                me->CastSpell(me, SPELL_EYE_VISUAL, true);
```
DoCastSelf

Keader commented on src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:
```diff
@@ -377,15 +377,17 @@ class npc_eye_of_acherus : public CreatureScript
                 init.SetFall();
                 me->GetMotionMaster()->LaunchMoveSpline(std::move(init), POINT_EYE_FALL, MOTION_PRIORITY_NORMAL, POINT_MOTION_TYPE);
+
+                _events.ScheduleEvent(EVENT_MOVE_START, 7000);
```
_events.ScheduleEvent(EVENT_MOVE_START, 7s);

Keader commented on src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:
```diff
@@ -400,9 +402,9 @@ class npc_eye_of_acherus : public CreatureScript
                         {
-                            DoCast(me, SPELL_EYE_FLIGHT_BOOST);
-
+                            me->CastSpell(me, SPELL_EYE_FLIGHT_BOOST);
```
DoCastSelf

Killyana committed:
DB: Fix "The Maiden's Fancy" spawns

Killyana committed:
DB/Creature: Plagued Snapper/Lake Snapper

Closes #22086

Jonne733 committed:
DB/Creature: Freed Sha'tar Warrior

Closes #22084

Jonne733 committed:
DB/Npc: Sand Gnome

Closes #22083

ZenoX92 committed:
DB/Quest: Patrol Schedules

Closes #22082

Alpharius committed:
DB/Quest: DE quests locales, Redridge/Stormwind

Closes #22080

Treeston committed:
Core/AI: Add a new LeavingWorld AI hook

Killyana committed:
DB/Quest: A Little Slime Goes a Long Way

Closes #20591

killerwife committed:
Core/Unit: Name and implement UNIT_FLAG_NON_ATTACKABLE_2 (PR #22087)

DanVS committed:
Fixes

DanVS committed:
Merge branch 'q12641' of https://github.com/DanVS/TrinityCore into q12641

Faq commented:
why this was closed?
